src/test/ui/borrowck/mut-borrow-of-mut-ref.rs

```diff
@@ -1,11 +1,11 @@
 // Suggest not mutably borrowing a mutable reference
+#![crate_type = "rlib"]
 
-fn main() {
-    f(&mut 0)
+pub fn f(b: &mut i32) {
+    g(&mut b);
+    //~^ ERROR cannot borrow
+    g(&mut &mut b);
```

Comment:
It would be nice to suggest removing `&mut &mut` here.
But I don't know how to do that.

Reply:
We could probably leave this for someone to fix it next time, this patch is already good I guess? The other thing I want to do is https://github.com/rust-lang/rust/issues/71389, related to this but seemed hard to me, need to check how to do it. @lzutao Do you have any idea roughly where needs to be change, otherwise I have to look into it myself.

Reply:
No ideal. I often grep for the part of error message in `compiler` dir to find relevant code emitted the errors.
Or use `-Z treat-err-as-bug` and `RUST_BACKTRACE` to find them.
I also need to enable `debug-logging` to play around with code.
Maybe estebank will have more mentor instructions if you ask him in the issue.

Reply:
I don't think it'd be easy to do this with the way that the code is currently organized, so we can probably leave this as a follow-up.